bucket.go

```diff
@@ -190,13 +190,11 @@ func (b *Bucket) CreateBucket(key []byte) (*Bucket, error) {
 // Returns an error if the bucket name is blank, or if the bucket name is too long.
 // The bucket instance is only valid for the lifetime of the transaction.
 func (b *Bucket) CreateBucketIfNotExists(key []byte) (*Bucket, error) {
-	child, err := b.CreateBucket(key)
-	if err == errors.ErrBucketExists {
-		return b.Bucket(key), nil
-	} else if err != nil {
-		return nil, err
+	child := b.Bucket(key)
```

Comment:
Side effect: When the db is opened in readonly mode, the caller can get a valid bucket if present. 

The existing implementation will return a `ErrTxNotWritable` error.

Reply:
Suggest not to change the behavior.

Reply:
Calling `CreateBucketIfNotExists` in read-only mode doesn't make sense. I don't think anybody relied on this behavior.

Reply:
This PR creates a hole (readonly transaction can even create bucket (actually already exist) ) which might cause unnecessary confusion to users. It can even be regarded as a minor harmless CVE.

Please add the following code at the beginning of the method,
```
	if b.tx.db == nil {
		return nil, ErrTxClosed
	} else if !b.tx.writable {
		return nil, ErrTxNotWritable
	} else if len(key) == 0 {
		return nil, ErrBucketNameRequired
	}
```

Reply:
Added.

Reply:
Also added benchmark results. See PR description. It doesn't help with the operation speed but reduces allocations when there are many buckets which is a valid use cases where buckets are dynamically generated.